server/queue/queue-activemq/src/main/java/org/apache/james/queue/activemq/metric/ActiveMQMetricCollectorImpl.java

```diff
@@ -0,0 +1,178 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.activemq.metric;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+import javax.jms.Connection;
+import javax.jms.ConnectionFactory;
+import javax.jms.JMSException;
+import javax.jms.MapMessage;
+import javax.jms.Message;
+import javax.jms.MessageConsumer;
+import javax.jms.MessageProducer;
+import javax.jms.Queue;
+import javax.jms.Session;
+import javax.jms.TemporaryQueue;
+
+import org.apache.james.metrics.api.GaugeRegistry;
+import org.apache.james.metrics.api.MetricFactory;
+import org.apache.james.queue.api.MailQueueName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import reactor.core.Disposable;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+public class ActiveMQMetricCollectorImpl implements ActiveMQMetricCollector {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ActiveMQMetricCollectorImpl.class);
+
+    public static final Duration REFRESH_DELAY = Duration.ofSeconds(2);
+    public static final Duration REFRESH_INTERVAL = Duration.ofSeconds(5);
+    public static final Duration RECEIVE_TIMEOUT = Duration.ofSeconds(1);
+    public static final Duration REFRESH_TIMEOUT = RECEIVE_TIMEOUT.multipliedBy(2);
```

Comment:
I think the intervals can be bigger (30s?)

We could imagine also enabling configuring them (ENV variables?)

Reply:
> I think the intervals can be bigger (30s?)

I think the interval of 5s seems ok, since the mean time to retrieve those metrics is about 50-80ms when James is not under load.

> We could imagine also enabling configuring them (ENV variables?)

Sounds good. How about `conf/activemq.properties` to streamline to other modules (e.g. healthcheck.properties, jvm.properties, etc.)

Reply:
If that is the way we go down to, we could take time to identify other properties of `EmbeddedActiveMQ` worth configuring...

Reply:
I am fine with this. I just wanted to keep this PR small. However, it totally makes sense to add some configuration options.

Reply:
Configuration option is available through conf/activemq.properties

Some thoughts about configuring `EmbeddedActiveMQ`:
- It might be possible to use an external ActiveMQ instance, but enable the embedded one by default `embedded=true`
- File locations for KahaDB and blobs